walkeeper/src/send_wal.rs

```diff
@@ -45,6 +46,9 @@ impl postgres_backend::Handler for SendWalHandler {
         } else if query_string.starts_with(b"START_REPLICATION") {
             ReplicationConn::new(pgb).run(self, pgb, &query_string)?;
             Ok(())
+        } else if query_string.starts_with(b"START_WAL_PUSH") {
+            ReceiveWalConn::new(pgb)?.run(self)?;
```

Comment:
Probably somewhere here (or in first lines of ReceiveWalConn::run()) you need to switch protocol to CopyBoth state like here: https://github.com/zenithdb/zenith/blob/main/pageserver/src/page_service.rs#L197

Reply:
Yup- I'd realized that shortly after I pushed this. Updated version coming soon, after I do a little bit more with a test C client to check that it works.